lib/hooks/csrf/index.js

```diff
@@ -77,8 +77,14 @@ module.exports = function(sails) {
           var allowCrossOriginCSRF = sails.config.csrf.origin.split(',').map(trim).indexOf(req.headers.origin) > -1;
 
           if (sails.config.csrf.protectionEnabled) {
-            var connect = require('express/node_modules/connect');
-
+            var connect;
+            try {
+              // Support npm v3 flat dependency installations.
+              connect = require('connect');
```

Comment:
1) Isn't this the express 4 connect? Does it work with express 3?
2) I think we'd also want to add this to `optionalDependencies` at least, otherwise things get confusing.